interfaces/builtin/desktop.go

```diff
@@ -622,9 +622,9 @@ func (iface *desktopInterface) AppArmorPermanentSlot(spec *apparmor.Specificatio
 
 // https://specifications.freedesktop.org/desktop-entry-spec/latest/file-naming.html
 // Desktop file id must be a valid D-Bus name:
-//  - A sequence of non-empty elements separated by dots
-//  - None of which starts with a digit
-//  - Each of which contains only characters from the set [A-Za-z0-9-_]
+//   - A sequence of non-empty elements separated by dots
+//   - None of which starts with a digit
+//   - Each of which contains only characters from the set [A-Za-z0-9-_]
```

Comment:
```suggestion
//   - Each of which contains only characters from the set [A-Za-z0-9_-]
```
Nitpick: As below, `-` in the set should be at the end so it's easy to tell if it's included, rather than used as a range indicator.

Reply:
Also, the regex below supports leading and trailing `-` and `_` characters... is that desired? This is probably more a question for @ZeyadYasser, sorry for nitpicking about this here :)

Reply:
it's a direct copy from the spec linked above https://specifications.freedesktop.org/desktop-entry-spec/latest/file-naming.html and AFAICT both will have the same effect. Also from regex(7):

```
       [...].  To include a literal '-', make it the first  or
       last character, or the second endpoint of a range.  [...]
```

so both notations are valid AFAIU

Reply:
> Also, the regex below supports leading and trailing `-` and `_` characters... is that desired? This is probably more a question for @ZeyadYasser, sorry for nitpicking about this here :)

Actually two lines below, There is a comment that asks the same question of whether we really want to support "-" since it is not even recommended by the spec, but supported. I left there as an open question.